Executor/Promise/Adapter/ReactPromiseAdapter.php

```diff
@@ -58,17 +58,20 @@ public function wait(Promise $promise)
         /** @var \React\Promise\PromiseInterface $reactPromise */
         $reactPromise = $promise->adoptedPromise;
 
+        $reactPromise->then(function ($values) use (&$resolvedValue, &$wait) {
+            $resolvedValue = $values;
+            $wait = false;
+        }, function ($reason) use (&$exception, &$wait) {
+            $exception = $reason;
+            $wait = false;
+        });
+
+        // wait until promise resolution
         while ($wait) {
-            $reactPromise->then(function ($values) use (&$resolvedValue, &$wait) {
```

Comment:
Maybe adding `usleep(1)` would make it less CPU intensive without sacrificing the performance